Buttons made visible in DevHub mobile view

kunalmohan opened the pull request:
Fixes #12594

New Add-on and Theme buttons were not visible in mobile view.
The issue is corrected by removing display none property and
changing flex direction of container to column for device-
width less than 960px.

before-
![Screenshot from 2019-11-07 15-10-25](https://user-images.githubusercontent.com/44079328/68377668-d6353e80-0170-11ea-9580-1033869fc6f4.png)

after-
![Screenshot from 2019-11-07 15-08-31](https://user-images.githubusercontent.com/44079328/68377683-dd5c4c80-0170-11ea-8915-1afe8221b10e.png)

kunalmohan commented:
I closed the request by mistake.

kunalmohan commented:
@kumar303 are there any issues with the PR because initially it did not pass some checks? Should I make any more changes?

Sorry for bringing this up so late :sweat_smile:

kumar303 commented:
Hi @kunalmohan. This has been in my review queue but I've had to give my time to other prioritized features. My apologies. I hope to get to it soon. Thanks for taking time to submit a patch.

kunalmohan commented:
Yeah, I could do that. I'll wait for your confirmation.

eviljeff commented:
I don't think it matters too much, but what @bobsilverberg said seems fine.  

Another option, if we think keeping them under one another at widths all the way up to 960px is too extreme, is to change the `@media` to `@media @medium` in `.DevHub-MyAddons-item-buttons` and `.DevHub-MyAddons-item-buttons-submit` (I've not experimented to see how this looks)

kunalmohan commented:
These are ss according to what @eviljeff is suggesting:
800px:
![800px](https://user-images.githubusercontent.com/44079328/70320493-e19c8800-184a-11ea-8055-b41e828571af.png)


715px:
![715px](https://user-images.githubusercontent.com/44079328/70320521-efeaa400-184a-11ea-86f7-fab9de76025a.png)

650px:
![650px](https://user-images.githubusercontent.com/44079328/70320541-ff69ed00-184a-11ea-887b-4b4f180fdc9e.png)


For <550px, it is the same as what I have implemented above.

muffinresearch commented:
> These are ss according to what @eviljeff is suggesting:
> 800px:
> ![800px](https://user-images.githubusercontent.com/44079328/70320493-e19c8800-184a-11ea-8055-b41e828571af.png)
> 
> 715px:
> ![715px](https://user-images.githubusercontent.com/44079328/70320521-efeaa400-184a-11ea-86f7-fab9de76025a.png)
> 
> 650px:
> ![650px](https://user-images.githubusercontent.com/44079328/70320541-ff69ed00-184a-11ea-887b-4b4f180fdc9e.png)
> 
> For <550px, it is the same as what I have implemented above.

Hi @kunalmohan This looks pretty good, if it's possible to make the 715px view match the layout shown in the 650px one that would be better and would avoid having the buttons floating in mid air. We'll also want to check that the state represented in the 715px above doesn't appear at some other point when reducing the width once that nit is fixed.

For < 550px the only other change I'd suggest is making the buttons wider to be 100% of the viewport width (less the default gutter either side). 

Let me know if you have any additional questions.

kunalmohan commented:
@muffinresearch I made the required changes. I had to introduce two new breakpoints, one of which I have introduced in the traditional CSS manner (I didn't want to make a new standard breakpoint that differs from an already defined one by just 30px).

bobsilverberg commented:
@kunalmohan Thanks for your continued efforts on this, and sorry for not getting back to you sooner. There shouldn't be a need to define a new breakpoint, nor to introduce a custom breakpoint. We should just try to make sure that things look good at the already established breakpoints.

kunalmohan commented:
Okay, I'll try to get the best results without defining any new breakpoints :sweat_smile:

kunalmohan commented:
> > These are ss according to what @eviljeff is suggesting:
> > 800px:
> > ![800px](https://user-images.githubusercontent.com/44079328/70320493-e19c8800-184a-11ea-8055-b41e828571af.png)
> > 715px:
> > ![715px](https://user-images.githubusercontent.com/44079328/70320521-efeaa400-184a-11ea-86f7-fab9de76025a.png)
> > 650px:
> > ![650px](https://user-images.githubusercontent.com/44079328/70320541-ff69ed00-184a-11ea-887b-4b4f180fdc9e.png)
> > For <550px, it is the same as what I have implemented above.
> 
> Hi @kunalmohan This looks pretty good, if it's possible to make the 715px view match the layout shown in the 650px one that would be better and would avoid having the buttons floating in mid air. We'll also want to check that the state represented in the 715px above doesn't appear at some other point when reducing the width once that nit is fixed.
> 
> For < 550px the only other change I'd suggest is making the buttons wider to be 100% of the viewport width (less the default gutter either side).
> 
> Let me know if you have any additional questions.

I have corrected the display at 715px. And I don't think it would be possible to make the buttons wider for viewport <550px without defining a new breakpoint there.

bobsilverberg commented:
> > > These are ss according to what @eviljeff is suggesting:
> > > 800px:
> > > ![800px](https://user-images.githubusercontent.com/44079328/70320493-e19c8800-184a-11ea-8055-b41e828571af.png)
> > > 715px:
> > > ![715px](https://user-images.githubusercontent.com/44079328/70320521-efeaa400-184a-11ea-86f7-fab9de76025a.png)
> > > 650px:
> > > ![650px](https://user-images.githubusercontent.com/44079328/70320541-ff69ed00-184a-11ea-887b-4b4f180fdc9e.png)
> > > For <550px, it is the same as what I have implemented above.
> > 
> > 
> > Hi @kunalmohan This looks pretty good, if it's possible to make the 715px view match the layout shown in the 650px one that would be better and would avoid having the buttons floating in mid air. We'll also want to check that the state represented in the 715px above doesn't appear at some other point when reducing the width once that nit is fixed.
> > For < 550px the only other change I'd suggest is making the buttons wider to be 100% of the viewport width (less the default gutter either side).
> > Let me know if you have any additional questions.
> 
> I have corrected the display at 715px. And I don't think it would be possible to make the buttons wider for viewport <550px without defining a new breakpoint there.

Thanks @kunalmohan. I see that the smallest breakpoint currently available is 700px. I was thinking you could define the buttons to be 100% width at <550px, as @muffinresearch suggested, but I also see that a breakpoint doesn't exist for that size. @muffinresearch can you suggest how that might be done without introducing a new breakpoint?

eviljeff commented:
The only sizes currently defined in devhub landing page are:
`<700px` (no `@media`, so "normal" size)
`>=700px` (`@medium`)
`>=960px` (`@large`)

Although it's possible to add extra breakpoints I spent some time tidying up unnecessary breakpoints previously so I'd prefer to stick with those tbh :)  Maybe we can have them max-width up to 700px?  Or just define a max-size directly on the button?

muffinresearch commented:
Looking at this I think it works better if you just remove the medium break-point change and make the `flex-direction: column;` for all breakpoints in `.DevHub-MyAddons-item-buttons`.

However, after doing this it really starts to remove the need for using flexbox, so instead you could simply ditch all the flex related properties on `.DevHub-MyAddons-item-buttons` and simply float right (left for RTL) `.DevHub-MyAddons-item-buttons-submit`. This then works better across multiple breakpoints.

After that the only change is to fix the button width to ~100%, and override that for larger viewports.

kunalmohan commented:
> Looking at this I think it works better if you just remove the medium break-point change and make the `flex-direction: column;` for all breakpoints in `.DevHub-MyAddons-item-buttons`.
> 
> However, after doing this it really starts to remove the need for using flexbox, so instead you could simply ditch all the flex related properties on `.DevHub-MyAddons-item-buttons` and simply float right (left for RTL) `.DevHub-MyAddons-item-buttons-submit`. This then works better across multiple breakpoints.
> 
> After that the only change is to fix the button width to ~100%, and override that for larger viewports.

Applying these changes, there is a range of viewport width which has the following layout:
![abc](https://user-images.githubusercontent.com/44079328/71104838-748dd880-21e2-11ea-8881-1b1fa90d14d3.png)
Here the width is 100% (for viewport smaller than `medium`).

For other viewports, layout looks good.

muffinresearch commented:
This still appears to be using flex-box. Note the part of the comment which refers to dropping that in favour of simple floats.

kunalmohan commented:
Nope, this does not use flexbox.
![abcd](https://user-images.githubusercontent.com/44079328/71256675-7c22be00-2357-11ea-8dda-2fe99560fc7c.png)

I suppose I could also remove all the margin properties from `.DevHub-MyAddons-item-buttons-submit` after using float.

muffinresearch commented:
> Nope, this does not use flexbox.
> ![abcd](https://user-images.githubusercontent.com/44079328/71256675-7c22be00-2357-11ea-8dda-2fe99560fc7c.png)
> 
> I suppose I could also remove all the margin properties from `.DevHub-MyAddons-item-buttons-submit` after using float.

I still see this for the diff: 

<img width="1072" alt="Buttons_made_visible_in_DevHub_mobile_view_by_kunalmohan_·_Pull_Request__12825_·_mozilla_addons-server_-_Firefox_Nightly" src="https://user-images.githubusercontent.com/1514/71816253-e6d33980-3079-11ea-9e7f-d986766c3b55.png">

Is a commit missing from the PR?

kunalmohan commented:
> Is a commit missing from the PR?

Yes, I had not pushed the latest changes. I have done so now.

muffinresearch commented:
Thanks for the patch @kunalmohan

caitmuenster commented:
Thanks so much for the patch, @kunalmohan! Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 

Would you be interested in creating an account on [mozillians.org](https://mozillians.org)? I'd be happy to vouch for you!

kunalmohan commented:
@caitmuenster Yes, I'd love that :)

kunalmohan commented:
@caitmuenster could you vouch for me? https://mozillians.org/en-US/u/kunalmohan99/

caitmuenster commented:
@kunalmohan You're vouched! 🎉 Welcome onboard! I look forward to seeing you around the project.
